pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -76,6 +82,40 @@ def _gen_repo():
         'notes': {'_repo-type': 'puppet-repo'},
     }
 
+class _CleanUpTestCaseMixin(object):
+    """Provide functions to clean up database and filesystem of Pulp"""
+
+    @classmethod
+    def reset_database(cls, server_config):
+        """Drop database and re-run migrations."""
+        cls.services = tuple((
+            cli.Service(cls.cfg, service) for service in _SERVICES
+        ))
+
+        # Stop all Pulp services
+        for service in cls.services:
+            service.stop()
+
+        # Drop database and re-run migrations
+        client = cli.Client(server_config, cli.echo_handler)
+        drop_db_command = ('mongo', 'pulp_database', '--eval', 'db.dropDatabase()')
+        client.run(drop_db_command)
+        migrations_command = ('sudo', '-u', 'apache', 'pulp-manage-db')
+        client.run(migrations_command)
```

Comment:
This can also be written as:

```python
        client.run("mongo pulp_database --eval 'db.dropDatabase()'".split())
        client.run('sudo -u apache pulp-manage-db')
```